### DISCUSSION TITLE
MeshTastic Android - OLC/PlusCode

### DISCUSSION THREAD

**Discussion by RiverMersey** [meshtastic/firmware#3030] [General]
Hi,

Last week I acquired 6 x HelTec v3 LoRa dev boards for review and experimentations.

Having researched the Android app menu, I was pleasantly surprised to notice the GPS coordinates has various options.

One such option, listed on the MeshTastics webpage as "OLS" alludes to this being equivalent to PlusCodes.

On selecting this option, the GPS format for my node continues to display as decimal.

HelTec v3 boards do not have on board GPS modules, however, the MeshTastic app does have a feature of making a cell-phone GPS data available to the dev board.

Passing the correct location data between the phone and the dev board does seem to work as expected.

Only the formatting of this data doesn't seem to work correctly using the OLC option.

PlusCodes does have a Git Repo which mathematically describes how to calculate and format from a decimal to a PlusCode - and visa versa.

As yet, I'm not sure where this calculation is done - on the dev board firmware or in the Android app?

Dev board firmware would consume precious clock-cycles and battery power, whereas Android app would have much more resources available.

Relaying the Android-calculated PlusCode back onto the dev board display would be highly useful - afaik, such a MeshTastic device would then be unique in being able to live calculate a (long-form) PlusCode on location. A short-form PlusCode would require access to a lookup database of words vs codes within the app.

Watching a GPS decimal reading change as the device moves will simultaneously alter 2 vectors. Also it might not be immediately clear to the observer what distance relationship there is between these 2 decimal GPS readings.

Watching a GPS long-form PlusCode device move around would only change the last digit for each move of about 3 metres. A pair of long-form PlusCode locations could be placed in a column and the observer reading from left to right would notice how similar the codes are digit to digit means how close the locations are.

For comparison, PlusCode calculation and short-form PlusCode words are all open source, whereas W3W (What 3 Words) is proprietary. Furthermore, W3W's word-list database seems to be extensive and cumbersome whereas PlusCodes can still be useful without any word database.

Essentially, what I'm asking is, could the OLC code be "tidied-up" - and better promoted(!)?

Doing so would make these types of dev boards - and the MeshTastic ecosystem - much more attractive commercially.

Such devices could then be mounted on vehicles, available for emergency services, construction, utility services and such like.

**Comment by garthvh** [comment]
Plus codes are an old feature that was removed quite a while ago, never got any use.

**Reply by RiverMersey** [reply]
Actually, PlusCodes are widely used, and unlike W3W are also Google Maps supported.

Even in "developed" countries such as here in UK, PlusCodes are very useful in the realms previously mentioned, additionally could be used for mapping grave-yards, farming-fields, etc.

With respect, I argue PlusCodes should be re-implemented in a revised way as they could be an ideal on a mesh network which provides relatively local access points.

For example, a macro or button on the messaging service could ping once or intervals the node location while the receiver(s) could display the received PlusCode followed by its own PlusCode.

Or, in the case that OLC/PlusCodes are to be deleted from the display menu - for the benefit of users new to the MeshTastic ecosystem, please actually delete all references to it in drop-down menus and on the website.
